Sources/Auth/Services/Signer/CacaoSigner.swift

```diff
@@ -0,0 +1,32 @@
+import Foundation
+
+protocol CacaoSignatureVerifying {
+    func verifySignature(_ cacao: Cacao) throws
+}
+
+struct CacaoSigner: CacaoSignatureVerifying {
+
+    enum Errors: Error {
+        case signatureValidationFailed
+    }
+
+    private let signer: Signer
+
+    init(signer: Signer) {
+        self.signer = signer
+    }
+
+    func sign(payload: CacaoPayload, privateKey: Data) throws -> CacaoSignature {
+        let message = try JSONEncoder().encode(payload) // TODO: SIWE encoding
+        let signature = try signer.sign(message: message, with: privateKey)
+        return CacaoSignature(t: "eip191", s: signature.toHexString(), m: String())
+    }
+
+    func verifySignature(_ cacao: Cacao) throws {
+        let sig = Data(hex: cacao.signature.s)
+        let message = try JSONEncoder().encode(cacao.payload)
```

Comment:
would it maybe better to add a new method to `SIWEMessageFormatting`
`formatMessage(from cacao: Cacao) -> String`?
for the implementation we can have a separate issue but we are not suppose to verify signature on cacao but on EIP-4361 message instead.